### PR TITLE
Bounties

### PR DESCRIPTION
**Pull request by auriium**
Gotta merge this quick so i can get back to work on kitpvp

### CODE REVIEW

File: core-authentication/pom.xml [hunk 1]
```diff
@@ -1,6 +1,28 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
     <modelVersion>4.0.0</modelVersion>
+    <build>
+        <plugins>
```
Comment (A248) [inline core-authentication/pom.xml]: You committed these everywhere.

File: mod-kitpvp/src/main/java/gg/solarmc/loader/kitpvp/KitPvp.java [hunk 2]
```diff
@@ -199,6 +199,31 @@ public void resetCurrentKillstreaks(Transaction transaction) {
         this.updateCurrentKillstreak(0);
     }
 
+    public void addBounty(Transaction transaction, int amount) {
```
Comment (A248) [inline mod-kitpvp/src/main/java/gg/solarmc/loader/kitpvp/KitPvp.java:224]: Needs javadoc

**Review by A248 — CHANGES_REQUESTED** [review]
Needs a couple of things then can be merged

File: core-authentication/pom.xml (hunk 1 above)
Comment (auriium) [reply core-authentication/pom.xml]: What?

File: mod-kitpvp/src/main/java/gg/solarmc/loader/kitpvp/KitPvp.java (hunk 2 above)
Comment (auriium) [reply mod-kitpvp/src/main/java/gg/solarmc/loader/kitpvp/KitPvp.java:224]: Fine

File: core-authentication/pom.xml (hunk 1 above)
Comment (auriium) [reply core-authentication/pom.xml]: Oops

Comment (auriium) [reply core-authentication/pom.xml]: Im not entirely sure how that happened

**Comment by auriium** [comment]
@A248

File: mod-kitpvp/src/main/java/gg/solarmc/loader/kitpvp/KitPvp.java [hunk 3]
```diff
@@ -199,6 +199,42 @@ public void resetCurrentKillstreaks(Transaction transaction) {
         this.updateCurrentKillstreak(0);
     }
 
+    /**
+     * Adds bounty currency to the player. Infallible.
+     *
+     * @param transaction transa
```
Comment (A248) [inline mod-kitpvp/src/main/java/gg/solarmc/loader/kitpvp/KitPvp.java]:
```suggestion
     * @param transaction the transaction
```

File: mod-kitpvp/src/main/java/gg/solarmc/loader/kitpvp/KitPvp.java [hunk 4]
```diff
@@ -199,6 +199,42 @@ public void resetCurrentKillstreaks(Transaction transaction) {
...
+     * @param amount amount to add
```
Comment (A248) [inline mod-kitpvp/src/main/java/gg/solarmc/loader/kitpvp/KitPvp.java:221]:
```suggestion
     * @param amount amount to add
     * @throws IllegalArgumentException if the amount is negative
```

**Review by A248 — COMMENTED** [review]
Don't you need a method to get the bounty?

**Comment by auriium** [comment]
@A248 merge it i dare you

**Comment by auriium** [comment]
FUCK WHYD IT CLOSE

File: mod-kitpvp/src/main/java/gg/solarmc/loader/kitpvp/KitPvp.java [hunk 5]
```diff
@@ -199,6 +199,58 @@ public void resetCurrentKillstreaks(Transaction transaction) {
         this.updateCurrentKillstreak(0);
     }
 
+    /**
+     * Gets the mounty
```
Comment (A248) [inline mod-kitpvp/src/main/java/gg/solarmc/loader/kitpvp/KitPvp.java]:
```suggestion
     * Gets the bounty
```

File: mod-kitpvp/src/main/java/gg/solarmc/loader/kitpvp/KitPvp.java [hunk 6]
```diff
@@ -199,6 +199,58 @@ public void resetCurrentKillstreaks(Transaction transaction) {
...
+     * @param transaction trans
```
Comment (A248) [inline mod-kitpvp/src/main/java/gg/solarmc/loader/kitpvp/KitPvp.java]:
```suggestion
     * @param transaction the transaction
```

File: mod-kitpvp/src/main/java/gg/solarmc/loader/kitpvp/KitPvp.java [hunk 7]
```diff
@@ -199,6 +199,58 @@ public void resetCurrentKillstreaks(Transaction transaction) {
...
+        int newValue = existingValue + amount;
+
+        statisticsRecord.setBounty(newValue);
+        statisticsRecord.store(KITPVP_STATISTICS.BOUNTY);
+
+        this.updateBounty(amount);
+    }
+
+    /**
+     * resets the bounty. infallible.
```
Comment (A248) [inline mod-kitpvp/src/main/java/gg/solarmc/loader/kitpvp/KitPvp.java]:
```suggestion
     * Resets the bounty. infallible.
```

File: mod-kitpvp/src/main/java/gg/solarmc/loader/kitpvp/OnlineKitPvp.java [hunk 8]
```diff
@@ -104,4 +111,8 @@ public int currentHighestKillstreaks() {
     public int currentExperience() {
         return experience;
     }
+
```
Comment (A248) [inline mod-kitpvp/src/main/java/gg/solarmc/loader/kitpvp/OnlineKitPvp.java:114]:
```suggestion

    /**
     * The cached bounty value. Not reliable
     * @return the bounty
     */
```

**Review by A248 — APPROVED** [review]

**Pull request merged**